### PR TITLE
fix(prices): update price list module to 2.x standards

### LINKED ISSUE #955: (Bug) Negative values can be set for price list items
Currently price lists can be set as values or as percentages. 

If set as a percentage this will be applied to the sale price of an inventory item, a negative percentage will reduce the price and a positive percentage will increase it.

If set as a value a negative value, this will be allowed by the client - this will be directly set at invoice. Although an invoice cannot be submitted with any value 0 or less this is still set. 

**Problem**
You can set a price list item as a value, less than 0 

**Expected Behavior**
Either the client or server will block a price list item being set as a value and less than 0

### LINKED ISSUE #967: (bug) Price List form buttons should use loading indicators and error notifications
The price list module is terribly out of date.  To ensure a better user experience, the submit buttons should be loading buttons and the error notifications should use `Notify.handleError` for consistency.

### PR DESCRIPTION
**Pull request by jniles**
This commit fixes the price list client, giving it Notify for error
handling, ui select for inventory items, and loading buttons. 
It also implements the client-side validation logic for price lists.

Importantly, this refuses to submit a negative number for the
price of an inventory item (though percentages can still be
negative).
 
Partially address #930.  Closes #955. Closes #967.

---
Thank you for contributing!

Before submitting this pull request, please verify that you have:
 - [x] Run your code through JSHint.  [Check out our styleguide](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/docs/STYLEGUIDE.md).
 - [x] Run integration tests.
 - [x] Run end-to-end tests.
 - [x] Accurately described the changes your are making in this pull request.

For a more detailed checklist, [see the official review checklist](https://docs.google.com/document/d/1nupLVLRXgSZJQo_acLgrwvPnN8RukfSiwRhSToj81uU/pub) that this PR will be evaluated against.

Ensuring that the above checkboxes are completed will help speed the review process and help build a stronger application.  Thanks!

### CODE REVIEW

**Comment by jniles** [comment]
@sfount can I get a review?

File: client/src/i18n/en.json [hunk 1]
```diff
@@ -1023,7 +1023,10 @@
     "PRICE_LIST_ITEMS" : "Price list items",
     "TITLE"            : "Price List Management",
     "UNABLE_TO_DELETE" : "Price list cannot be deleted - it may currently be assigned to patient/ debtor groups",
-    "UPDATE"           : "Update a Price List"
+    "UPDATE"           : "Update a Price List",
+    "ERRORS" : {
+      "HAS_NEGATIVE_PRICE" : "You cannot assign a negative price to an item.  Please either assign a positive percentage, a negative percentage, or a positive value."
```
Comment (sfount) [inline client/src/i18n/en.json:1033]: We should collectively introduce the French translation for this key before it is used in a production environment.

File: client/src/partials/price_list/modal.html [hunk 2]
```diff
@@ -13,36 +13,53 @@
...
 
     <div class="form-group" ng-class="{ 'has-error' : ModalForm.$submitted && ModalForm.inventory_uuid.$invalid }">
       <label class="control-label">{{ "FORM.LABELS.INVENTORY" | translate }}</label>
-      <select class="form-control" name="inventory_uuid" ng-model="ModalCtrl.data.inventory_uuid" id="inventory_uuid" ng-options="i.uuid as i.label for i in ModalCtrl.inventories | orderBy:'label'" required>
-        <option value="" disabled> -- {{ 'FORM.SELECT.INVENTORY' | translate }} -- </option>
-      </select>
+      <ui-select
+        name="inventory_uuid"
+        ng-model="ModalCtrl.data.inventory_uuid"
+        ng-model-options="{ 'debounce' : 200  }"
```
Comment (sfount) [inline client/src/partials/price_list/modal.html]: This should be removed or addressed here https://github.com/IMA-WorldHealth/bhima-2.X/issues/972.

File: client/src/partials/price_list/modal.js [hunk 3]
```diff
@@ -2,23 +2,25 @@ angular.module('bhima.controllers')
...
-  // load Inventory 
-  Inventory.getInventoryItems()
-  .then(function (data) {
-    vm.inventories = data;
-  }).catch(handler);
+  // load Inventory
+  Inventory.read()
+    .then(function (inventory) {
+      vm.inventories = inventory;
+      console.log('inventory:', inventory);
```
Comment (sfount) [inline client/src/partials/price_list/modal.js]: This can be removed.

File: client/src/partials/price_list/pricelist.html [hunk 4]
```diff
@@ -2,14 +2,14 @@
   <div class="bhima-title">
     <ol class="headercrumb">
       <li class="static">{{ "TREE.ADMIN" | translate }}</li>
-      <li class="active">{{ "PRICE_LIST.TITLE" | translate }}</li>
+      <li class="title">{{ "PRICE_LIST.TITLE" | translate }}</li>
```
Comment (sfount) [inline client/src/partials/price_list/pricelist.html:5]: 👍

**Review by sfount — COMMENTED** [review]
Changes LGTM. Using up to date features (`bh-loading` etc.) will make these pages much more reliable. 

There are some minor comments that could be discussed.

File: client/src/partials/price_list/modal.js (hunk 3 above)
Comment (jniles) [reply client/src/partials/price_list/modal.js]: Whoops!  Good catch.

File: client/src/partials/price_list/modal.html (hunk 2 above)
Comment (jniles) [reply client/src/partials/price_list/modal.html]: Sure.

File: client/src/i18n/en.json (hunk 1 above)
Comment (jniles) [reply client/src/i18n/en.json:1033]: Good catch :+1:

**Comment by jniles** [comment]
@sfount, changes made!

**Comment by jniles** [comment]
@mbayopanda, since @sfount is not feeling well, can you do a review of this?

**Review by sfount — APPROVED** [review]
Changes LGTM.

**Pull request merged**